guides/source/contributing_to_ruby_on_rails.md

```diff
@@ -485,6 +485,7 @@ You can invoke `test_jdbcmysql`, `test_jdbcsqlite3` or `test_jdbcpostgresql` als
 ### Warnings
 
 The test suite runs with warnings enabled. Ideally, Ruby on Rails should issue no warnings, but there may be a few, as well as some from third-party libraries. Please ignore (or fix!) them, if any, and submit patches that do not issue new warnings.
+An important note is that while developing locally it is helpful to set `RAILS_STRICT_WARNINGS` to replicate Rails CI behavior. This ensures that there are no surprise failures in the CI results for a contributor's pull request.
```

Comment:
"This ensures that there are no surprise failures" is not clear what 'surprise failures' means.
Would you clarify that by setting `RAILS_STRICT_WARNINGS`, the test suite fails if warnings are raised?